### PR TITLE
Bug fix for Object.toCanvasElement()

### PR DESCRIPTION
**Pull request by GreenFootballs**
Replaces the bounding rect calculation when an object has a shadow.

### CODE REVIEW

**Comment by GreenFootballs** [comment]
The node visual unit tests appear to be failing because of the same bug this commit tries to fix - it's getting a different size for a text object with a shadow.

File: src/shapes/object.class.js [hunk 1]
```diff
@@ -1711,30 +1711,49 @@
...
           boundingRect = this.getBoundingRect(true, true),
-          shadow = this.shadow, scaling,
-          shadowOffset = { x: 0, y: 0 }, shadowBlur,
-          width, height;
+          shadow = this.shadow;
 
       if (shadow) {
-        shadowBlur = shadow.blur;
-        if (shadow.nonScaling) {
-          scaling = { scaleX: 1, scaleY: 1 };
```
Comment (asturur) [inline src/shapes/object.class.js:1723]: In the new code we are missing both the nonScaling attribute effect from the shadow and the getObjectScaling usage that will count for the groups

**Comment by asturur** [comment]
UTs fail because the dataURL code changed behaviour, so that may be normal. And we need to update the tests.
I'm just not convinced this code is entirely better than the old one.

**Comment by GreenFootballs** [comment]
OK, got it. The nonScaling attribute is a simple fix, but accounting for groups and active selections will take a bit more work because it needs to find an overall bounding box that includes the shadows of each element. I assume I should refrain from using ES5/ES6 methods like Array.reduce()?

**Comment by asturur** [comment]
I'll get to this PR soon

**Comment by GreenFootballs** [comment]
Here's another try. This one accounts for groups and active selections that contain objects with shadows by looping over the objects, getting the extended bounding box for each one and storing it in an array, then using the array to calculate the total box size.

The only thing I wasn't sure about is whether it's OK to use a forEach loop, or should I convert that to an old-school for loop?

**Comment by asturur** [comment]
no this is a slow call anyway. This is not in the render hot loop. ForEach is the preferred way.

**Comment by GreenFootballs** [comment]
Here's a Codepen example of this code - you can toggle the original toCanvasElement() in and out to see the difference.

https://codepen.io/green_footballs/pen/VwmNeWQ?editors=1011

**Comment by asturur** [comment]
Closing this PR because we found out were the issue is.

**Pull request closed without merging**